plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PassThrough } from 'stream';
+import { Logger } from 'winston';
+import * as winston from 'winston';
+import { JsonValue } from '@backstage/config';
+import { TaskBroker, Task } from './types';
+import fs from 'fs-extra';
+import path from 'path';
+import { TemplateActionRegistry } from './TemplateConverter';
+
+type Options = {
+  logger: Logger;
+  taskBroker: TaskBroker;
+  workingDirectory: string;
+  actionRegistry: TemplateActionRegistry;
+};
+
+export class TaskWorker {
+  constructor(private readonly options: Options) {}
+
+  start() {
+    (async () => {
+      for (;;) {
+        const task = await this.options.taskBroker.claim();
+        await this.runOneTask(task);
+      }
+    })();
+  }
+
+  async runOneTask(task: Task) {
+    try {
+      const { actionRegistry, logger } = this.options;
+
+      const workspacePath = path.join(
+        this.options.workingDirectory,
+        await task.getWorkspaceName(),
+      );
+      await fs.ensureDir(workspacePath);
+
+      const taskLogger = winston.createLogger({
+        level: process.env.LOG_LEVEL || 'info',
+        format: winston.format.combine(
+          winston.format.colorize(),
+          winston.format.timestamp(),
+          winston.format.simple(),
+        ),
+        defaultMeta: {},
+      });
+
+      const stream = new PassThrough();
+      stream.on('data', data => {
+        const message = data.toString().trim();
+        if (message?.length > 1) task.emitLog(message);
+      });
+
+      taskLogger.add(new winston.transports.Stream({ stream }));
+
+      // Give us some time to curl observe
+      task.emitLog('Task claimed, waiting ...');
+      await new Promise(resolve => setTimeout(resolve, 5000));
+      task.emitLog(`Starting up work with ${task.spec.steps.length} steps`);
```

Comment:
as a log reader it would be nice to see the actual steps' specs, and the workspace name etc. maybe on yaml form on multiple lines, at least in debug mode, otherwise at least the list of action ids or so